src/main/java/gvs/ui/logic/app/AppViewModel.java

```diff
@@ -41,15 +52,58 @@
   private static final String PROMT_MESSAGE = "no active session";
   private static final Logger logger = LoggerFactory
       .getLogger(AppViewModel.class);
+  
 
-  // TODO: do we still need the persistor here? @mtrentini
+  // TODO: do we still need the persistor here?
   public AppViewModel(ApplicationModel appModel,
-      ApplicationController appController, IPersistor persistor) {
+      ApplicationController appController, IPersistor persistor,
+      BorderPane rootLayout) {
     this.appModel = appModel;
     this.appModel.addObserver(this);
     this.appController = appController;
     this.persistor = persistor;
     this.currentSessionName.set(PROMT_MESSAGE);
+    this.rootLayout = rootLayout;
+    sessionNames
+        .addListener((ListChangeListener.Change<? extends String> c) -> {
+          if (!sessionIsInitialized) {
+            initSessionLayout();
+          }
+          if (sessionNames.size() == 1) {
+            displaySession();
+          } else if (sessionNames.isEmpty()) {
+            hideSession();
+          }
+        });
+  }
+
+  private void initSessionLayout() {
+    logger.info("Initializing session layout.");
+    FXMLLoader loader = new FXMLLoader();
```

Comment:
Use Dependency Injection: 
`  private GuiceContext context = new GuiceContext(this, () -> Arrays.asList(new GuiceBaseModule()));`

and 

`@Inject
  private FXMLLoader fxmlLoader;`